### PR TITLE
F22 p5 release

### PR DESCRIPTION
**Pull request by MattyMay**
Bump requirements and update dockerfile.

I also added `pip install --upgrade pip setuptools wheel` to the other project Dockerfiles so that doesn't get forgotten later.

### CODE REVIEW

File: p1/Dockerfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ USER autograder:autograder
 
 COPY requirements.txt .
 RUN $PYTHON -m venv env-p1
+RUN env-p1/bin/pip install --upgrade pip setuptools wheel
```
Comment (awdeorio) [inline p1/Dockerfile:8]: Great idea

File: p5/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-astroid==2.11.1
+astroid==2.12.12
```
Comment (awdeorio) [inline p5/requirements.txt:1]: This file doesn't seem to match the same file in https://github.com/eecs485staff/p5-search-engine/pull/634/files I'm not sure which one is right.

**Review by awdeorio — CHANGES_REQUESTED** [review]
Found a couple possible problems

File: p5/requirements.txt (hunk 2 above)
Comment (MattyMay) [reply p5/requirements.txt:1]: Oops, I must've copied it over from the wrong branch. Should be good now.

File: p1/Dockerfile (hunk 1 above)
Comment (MattyMay) [reply p1/Dockerfile:8]: I already made the change for P4 in #30

**Review by awdeorio — APPROVED** [review]
LGTM

**Pull request merged**